addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4550,10 +4550,16 @@ var SnippetsMenu = Widget.extend({
             // TODO In future version use tool method to determine isMobilePreview.
             const mobileViewThreshold = MEDIAS_BREAKPOINTS[SIZES.LG].minWidth;
             const isMobilePreview = this.$body[0].ownerDocument.defaultView.frameElement.clientWidth < mobileViewThreshold;
+            const selection = this.options.wysiwyg.odooEditor.document.getSelection();
+            const selectedSection = selection.getRangeAt(0).startContainer.parentElement.closest("section");
```

Comment:
The document selection is not always the snippet with the overlay. For example, in this screenshot:

![image](https://github.com/user-attachments/assets/53bf1710-eea3-48d4-adc9-94dbf1f50bcc)

The selection includes the text "Useful Links" which is below the selected snippet. In this case, the fix doesn’t work. I think we need to find another way to get the selected snippet.

Reply:
hello @bvr-odoo 
```const selectedSection = this.options.wysiwyg?.lastElement.closest("section");```  this one seems to be working with last suggestion when selection and overlay is different, ```getSelection()``` does not give always the correct answer for our use.
although I was just informing, test when it's open to review.

Reply:
> The document selection is not always the snippet with the overlay. For example, in this screenshot:
> 
> ![image](https://private-user-images.githubusercontent.com/52911687/382756544-53bf1710-eea3-48d4-adc9-94dbf1f50bcc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzMDU2ODQsIm5iZiI6MTczMTMwNTM4NCwicGF0aCI6Ii81MjkxMTY4Ny8zODI3NTY1NDQtNTNiZjE3MTAtZWVhMy00OGQ0LWFkYzktOTRkYmYxZjUwYmNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTExVDA2MDk0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4ZTEwYjU4MWQwZTNiYTZhNTg5NjI2ZWNjY2ZjYzg5YzBlZjZhOWNkNzc5NWE3Yzk0NWJlZmM5OTQ2MmJiNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4p93fqmO_2FQuxpn69ux55K8eQ2pFAQBMR9sWcNNMcI)
> 
> The selection includes the text "Useful Links" which is below the selected snippet. In this case, the fix doesn’t work. I think we need to find another way to get the selected snippet.

@bvr-odoo This looks issue in Mobile, If we are selecting some text then current Snippet should be unselected, for instance the solution given by @rahg-odoo will work here but it looks issue in mobile screen/preview.